src/lib/commands.js

```diff
@@ -0,0 +1,274 @@
+			...data,
+			...params
+		}, (metadata) => {
+			const consent = {
+				metadata,
+				gdprApplies: config.gdprApplies,
+				hasGlobalScope: config.storeConsentGlobally,
+				...data,
+				vendorListVersion: params.vendorListVersion
+			};
+
+			callback(consent, true);
+		});
+	};
+
+	/**
+	 * Generates consent object based on input params: consents and vendor list
+	 */
+	const getConsentObject = (callback, consents, vendorList) => {
+		if (!consents || !vendorList) {
+			return callback({}, false);
+		}
+
+		const { persistedConsentData } = store;
+
+		const tcModel = new TCModel(new GVL(vendorList));
+		tcModel.cookieVersion = COOKIE_VERSION;
+		tcModel.cmpId = CMP_ID;
+		tcModel.cmpVersion = CMP_VERSION;
+		tcModel.created = persistedConsentData && persistedConsentData.created || new Date();
+		tcModel.lastUpdated = persistedConsentData && persistedConsentData.lastUpdated || new Date();
+		tcModel.vendorListVersion = tcModel.vendorListVersion_ = vendorList.vendorListVersion;
+		tcModel.isServiceSpecific = true;
+		tcModel.supportOOB = false;
+
+		for (let key of Object.keys(consents)) {
+			if (tcModel.hasOwnProperty(key)) {
+				tcModel[key].set(consents[key]);
+			}
+		}
+
+		const encoded = encodeConsentData(tcModel);
+
+		const {
+			cmpId,
+			cmpVersion,
+			policyVersion: tcfPolicyVersion,
+			supportOOB,
+			vendorListVersion,
+			isServiceSpecific,
+			useNonStandardStacks,
+			publisherCountryCode: publisherCC,
+			purposeConsents,
+			purposeLegitimateInterests,
+			publisherConsents,
+			publisherLegitimateInterests,
+			vendorConsents,
+			vendorLegitimateInterests,
+			vendorsDisclosed,
+			vendorsAllowed,
+			specialFeatureOptIns,
+			created,
+			lastUpdated
+		} = tcModel;
+
+		callback({
+			tcString: encoded,
+			cmpId,
+			cmpVersion,
+			tcfPolicyVersion,
+			supportOOB,
+			vendorListVersion,
+			isServiceSpecific,
+			useNonStandardStacks,
+			publisherCC,
+			created,
+			lastUpdated,
+			gdprApplies: config.gdprApplies,
+			outOfBand: {
+				allowedVendors: vectorToObject(vendorsAllowed),
+				disclosedVendors: vectorToObject(vendorsDisclosed)
+			},
+			purpose: {
+				consents: vectorToObject(purposeConsents),
+				legitimateInterests: vectorToObject(purposeLegitimateInterests)
+			},
+			vendor: {
+				consents: vectorToObject(vendorConsents),
+				legitimateInterests: vectorToObject(vendorLegitimateInterests)
+			},
+			specialFeatureOptins: vectorToObject(specialFeatureOptIns),
+			publisher: {
+				consents: vectorToObject(publisherConsents),
+				legitimateInterests: vectorToObject(publisherLegitimateInterests),
+				customPurposes: {
+					purposes: {},
+					legitimateInterests: {}
+				},
+				restrictions: {},
+			},
+			listenerId: undefined
+		}, true);
+	};
+
+
+	/**
+	 * Trigger the consent tool UI to be shown
+	 */
+	const showConsentTool =  (callback = () => {}) => {
+		store.updateSection();
+		store.toggleConsentToolShowing(true);
+
+		log.info(`Notify event: openConsentTool`);
```

Comment:
...and also I think that we will need some flag like `openConsentTool` to immediately dispatch listeners when someone wants to call `addEventListener` while consent tool is already showing

Reply:
Done